src/components/exchange/AvailableNetworksv2.tsx

```diff
@@ -0,0 +1,233 @@
+import lang from 'i18n-js';
+import React, { useCallback, useMemo } from 'react';
+import RadialGradient from 'react-native-radial-gradient';
+import Divider from '../Divider';
+import { CoinIcon } from '../coin-icon';
+import ChainBadge from '../coin-icon/ChainBadge';
+import { Box, Inline, Text } from '@/design-system';
+import networkInfo from '@/helpers/networkInfo';
+import { useNavigation } from '@/navigation';
+import { ETH_ADDRESS, ETH_SYMBOL } from '@/references';
+import Routes from '@/navigation/routesNames';
+import { position } from '@/styles';
+import { ethereumUtils } from '@/utils';
+import { CurrencySelectionTypes, ExchangeModalTypes, Network } from '@/helpers';
+import { useSwapCurrencyHandlers } from '@/hooks';
+import { AssetType, RainbowToken } from '@/entities';
+import { useTheme } from '@/theme';
+import { EthereumAddress } from '@rainbow-me/swaps';
+import { ButtonPressAnimation } from '../animations';
+import ContextMenuButton from '@/components/native-context-menu/contextMenu';
+
+const NOOP = () => null;
+
+type implementation = { address: EthereumAddress; decimals: number };
```

Comment:
wondering what `implementation` means, is used for `networks`, is the asset object for the specific network?

Reply:
ah i found the same type on `useAdditionalAssetData`, could you add the type there as well?

Reply:
i saw you updated `useAdditionalAssetData` with this same type, i meant that would be good adding it to entities or importing it from one file

Reply:
got it, just changed